lib/plugins/amplitudePlugin.spec.ts

```diff
@@ -0,0 +1,134 @@
+import * as amplitude from '@amplitude/analytics-node'
+import type { BaseEvent, EnrichmentPlugin, Event } from '@amplitude/analytics-types'
+import { buildClient, sendGet } from '@lokalise/node-core'
+import type { FastifyInstance } from 'fastify'
+import fastify from 'fastify'
+
+import { amplitudePlugin, amplitudeTrack } from './amplitudePlugin'
+
+describe('amplitudePlugin', () => {
+  let app: FastifyInstance
+
+  beforeEach(() => {
+    app = fastify()
+  })
+
+  afterEach(async () => {
+    await app.close()
+  })
+
+  it('skip initialization if isEnabled is false', async () => {
+    const initSpy = jest.spyOn(amplitude, 'init')
+
+    await app.register(amplitudePlugin, {
+      isEnabled: false,
+    })
+
+    expect(initSpy).not.toHaveBeenCalled()
+  })
+
+  it('throw error if api key is not defined', async () => {
+    const initSpy = jest.spyOn(amplitude, 'init')
+
+    let error = null
+    try {
+      await app.register(amplitudePlugin, {
+        isEnabled: true,
+      })
+    } catch (e) {
+      error = e
+    }
+
+    expect(initSpy).not.toHaveBeenCalled()
+    expect(error).not.toBeNull()
+  })
+
+  it('Basic initialization', async () => {
+    const initSpy = jest.spyOn(amplitude, 'init')
+    const addHookSpy = jest.spyOn(app, 'addHook')
+
+    await app.register(amplitudePlugin, {
+      isEnabled: true,
+      apiKey: 'This is an api key',
+    })
+
+    expect(initSpy).toHaveBeenCalled()
+    expect(addHookSpy).toHaveBeenCalledTimes(0)
+  })
+
+  it('initialization tracking api usage', async () => {
+    const initSpy = jest.spyOn(amplitude, 'init')
+    const addHookSpy = jest.spyOn(app, 'addHook')
+
+    await app.register(amplitudePlugin, {
+      isEnabled: true,
+      apiKey: 'This is an api key',
+      apiUsageTracking: (): BaseEvent => ({
+        event_type: 'My test event',
+      }),
+    })
+
+    expect(initSpy).toHaveBeenCalled()
+    expect(addHookSpy).toHaveBeenCalled()
+  })
+
+  it('amplitude track', async () => {
+    const trackSpy = jest.spyOn(amplitude, 'track')
+    await app.register(amplitudePlugin, {
+      isEnabled: true,
+      apiKey: 'This is an api key',
+    })
+
+    const event: BaseEvent = { event_type: 'event tracked' }
+    amplitudeTrack(event)
+
+    expect(trackSpy).toHaveBeenCalledWith(event)
+  })
+
+  it('api usage tracking', async () => {
+    const trackSpy = jest.spyOn(amplitude, 'track')
+    const event: BaseEvent = { event_type: 'My api event' }
+    await app.register(amplitudePlugin, {
+      isEnabled: true,
+      apiKey: 'This is an api key',
+      apiUsageTracking: (): BaseEvent => event,
+    })
+    await app
```

Comment:
you can simplify this code, fastify comes with built-in HTTP request testing support, via app.inject - https://github.com/fastify/fastify/blob/main/docs/Guides/Testing.md

You don't need to listen at all, and you don't need an http client built, you can inject requests directly into app instance. It's also faster.

Reply:
I didn't know it! it is really useful 😍, thank you so much